### PR TITLE
#30 upgrade cards

### PR DESCRIPTION
**Pull request by rafaeljpc**

### CODE REVIEW

File: model/actor.gd [hunk 1]
```diff
@@ -71,6 +74,18 @@ func step_time():
 	if can_draw():
 		draw_cooldown -= draw_rate
 
+func set_upgrade(upgrade):
+	if upgrades.size() == upgrade_slot:
+		upgrades.pop_front();
```
Comment (Kazuo256) [inline model/actor.gd]: The expected behavior from the design document was to not allow new upgrades if there are no slots available.

File: scenes/core/cards.gd [hunk 2]
```diff
@@ -1,7 +1,11 @@
 
 extends Node
 
+const Card = preload("res://model/card_ref.gd")
+
 func get_card_id(card):
+	if not card extends Card:
+		assert(false)
```
Comment (Kazuo256) [inline scenes/core/cards.gd]: I believe this lines can be simplified to:
```gdscript
assert(card extends Card)
```

**Comment by rafaeljpc** [comment]
Done

File: model/actor.gd (hunk 1 above)
Comment (Kazuo256) [inline model/actor.gd]: What about this part? XD

Comment (rafaeljpc) [inline model/actor.gd]: I asked you at dircord about it....
I thought that is was possible to renew the upgrades until we make a better management upgrade system (or interface)

Comment (rafaeljpc) [inline model/actor.gd]: According with our conversation I updated the code

**Pull request merged**